### PR TITLE
PP-4937 Add Stripe Account endpoint

### PR DESCRIPTION
**Pull request by DanailMinchev**
## WHAT YOU DID

Add `/v1/api/accounts/{accountId}/stripe-account` endpoint which can be used to query the `stripe_account_id` stored in `GatewayAccount` credentials.

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/gatewayaccount/model/StripeAccountResponse.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package uk.gov.pay.connector.gatewayaccount.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class StripeAccountResponse {
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/model/StripeAccountResponse.java:9]: I’d be very tempted to make this pseudo-immutable:

```java
public class StripeAccountResponse {

  @JsonProperty("stripe_account_id")
  private final String stripeAccountId;

  public StripeAccountResponse(String stripeAccountId) {
    this.stripeAccountId = Objects.requireNonNull(stripeAccountId)
  }

}
```

File: src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+import org.apache.commons.lang3.StringUtils;
+import uk.gov.pay.connector.gatewayaccount.model.GatewayAccountEntity;
+import uk.gov.pay.connector.gatewayaccount.model.StripeAccountResponse;
+
+import java.util.Optional;
+
+public class StripeAccountService {
+
+    public Optional<StripeAccountResponse> newStripeAccountResponse(GatewayAccountEntity gatewayAccountEntity) {
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java]: Not convinced by the name… `getStripeAccountResponse`, maybe?

File: src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        if (gatewayAccountEntity.getCredentials() == null ||
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java]: Can `getCredentials()` ever actually return `null`?

File: src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+                gatewayAccountEntity.getCredentials().isEmpty() ||
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java]: We don’t need the `isEmpty()` check; `get("stripe_account_id")` will return `null`, which the code handles

File: src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+                StringUtils.isEmpty(gatewayAccountEntity.getCredentials().get("stripe_account_id"))) {
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java]: I wouldn’t worry about blank strings, just `null`

File: src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+            return Optional.empty();
+        }
+
+        StripeAccountResponse stripeAccountResponse = new StripeAccountResponse();
+        stripeAccountResponse.setStripeAccountId(gatewayAccountEntity.getCredentials().get("stripe_account_id"));
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java]: We can probably avoid doing `gatewayAccountEntity.getCredentials().get("stripe_account_id"))` twice.

If my other suggestions are implemented, we could do something like this:

```java
return Optional.ofNullable(gatewayAccountEntity.getCredentials())
        .map(credentials -> credentials.get("stripe_account_id"))
        .map(StripeAccountResponse::new);
```

File: src/test/java/uk/gov/pay/connector/it/resources/StripeAccountResourceITest.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+                .get("/v1/api/accounts/" + testAccount.getAccountId() + "/stripe-account")
+                .then()
+                .statusCode(200)
+                .body("stripe_account_id", is(STRIPE_ACCOUNT_ID));
+    }
+
+    @Test
+    public void getStripeAccountReturnsNotFoundResponseWhenGatewayAccountNotFound() {
+        givenSetup()
+                .get("/v1/api/accounts/1337/stripe-account")
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/it/resources/StripeAccountResourceITest.java:38]: 😀

File: src/test/java/uk/gov/pay/connector/it/resources/StripeAccountResourceITest.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+                .insert();
+
+        givenSetup()
+                .get("/v1/api/accounts/" + testAccount.getAccountId() + "/stripe-account")
+                .then()
+                .statusCode(404);
+    }
+
+    @Test
+    public void getStripeAccountReturnsNotFoundResponseWhenGatewayAccountCredentialsAreEmpty() {
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/it/resources/StripeAccountResourceITest.java:58]: This is covered by the `StripeAccountServiceTest`: I wouldn’t retest it here

**Review by alexbishop1 — DISMISSED** [review]
This is good. I’ve left a few suggestions but nothing big enough to block approval.

File: src/test/java/uk/gov/pay/connector/it/resources/StripeAccountResourceITest.java (hunk 8 above)
Comment (alexbishop1) [reply src/test/java/uk/gov/pay/connector/it/resources/StripeAccountResourceITest.java:58]: Actually, I’ve changed my mind about this; I now see that this test is the only test that checks what the resource does if the `StripeAccountService` returns an empty `Optional`. Keep it.

File: src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java (hunk 2 above)
Comment (DanailMinchev) [reply src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java]: We discussed to use `buildStripeAccountResponse`, thanks

File: src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java (hunk 3 above)
Comment (DanailMinchev) [reply src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java]: I am not sure, the field of the class is `null`, that is why I decided to check for `null`s, as discussed I will check for `null`s

File: src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java (hunk 4 above)
Comment (DanailMinchev) [reply src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java]: Good point!

File: src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java (hunk 6 above)
Comment (DanailMinchev) [reply src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java]: Beautiful! 👍

**Review by alexbishop1 — DISMISSED** [review]

**Review by alexbishop1 — APPROVED** [review]

**Pull request merged**